### DISCUSSION TITLE
Mac Build, application xml format

### DISCUSSION THREAD

**Discussion by Kelsonk** [airsdk/Adobe-Runtime-Support#2629] [General]
I'm currently going through the mess of building and uploading a Mac app to the App Store. I've gone through all sorts of issues, but I currently think there's something wrong with my application xml. I don't believe any of the keys that I'm setting are going into the info.plist file. To start, I pretty much copied my application xml from my iOS build of the app to build the new Mac one.

I noticed all the entitlement and infoadditions is underneath the <iPhone> tag, does this need to be changed? I can't find documentation anywhere.

I currently have an entitlements file that I'm signing the app with, is this necessary? Can I just throw the entitlements under the <entitlements> flag in the app xml? Finding docs on this has been difficult, any help would be greatly encourgae (or even a default application.xml for Mac builds?)

**Comment by Kelsonk** [comment]
I followed this guide, I think I fixed my issues.  Sheesh, what a nightmare.

https://forum.starling-framework.org/d/23731-create-a-mac-app-ready-to-submit-to-the-mac-store/17

**Reply by ajwfrost** [reply]
Hi

That guide looks pretty good; in terms of the signing process, it is possible to do this via the entitlements in the application descriptor (under the macOS section) but only if you sign the application using ADT.

You can either do this in two steps:

1. Create your bundle:
`adt -package [air signing info] -target bundle AppName.app app.xml Main.swf ....`
2. Sign it:
`adt -sign -storetype KeychainStore -alias "Developer ID Application: name (team)" -target bundle AppName.app SignedAppName.app`

Or you can do it all in one step, by providing both sets of signing information on the command line (similarly this works on Windows i.e. the first set of signing options is used for the AIR signatures.xml, and the second set of signing options are the 'native' signing of the resulting package..)

`adt -package [air signing info] -target bundle -storetype KeychainStore -alias "Developer ID Application: name (team)" AppName.app app.xml Main.swf ....`

In these cases, any entitlements within the app descriptor `<macOS><Entitlements>` section will get added (as well as some of the default ones that are required for all AIR apps).


FYI there is some information on this at:
https://airsdk.dev/docs/tutorials/platform/macos/creating-macos-pkg-files

**Comment by Kelsonk** [comment]
I have never seen a guide anywhere describe the existence of a <macOS> tag in the application xml. I finally got this working, but it's by editing the Info.plist manually, and signing the app with a seperate entitilements file.
